sdk/internal/test/tracing/validator.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package tracing
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+	"github.com/Azure/azure-sdk-for-go/sdk/tracing/azotel"
+	"github.com/stretchr/testify/require"
+)
+
+// NewSpanValidator creates a Provider that verifies a span was created that matches the specified SpanMatcher.
+// The returned Provider can be used to create a client with a tracing provider that will validate spans in unit tests.
+func NewSpanValidator(t *testing.T, matcher SpanMatcher) tracing.Provider {
+	return tracing.NewProvider(func(name, version string) tracing.Tracer {
+		tt := matchingTracer{
+			matcher: matcher,
+		}
+
+		t.Cleanup(func() {
+			require.NotNil(t, tt.match, "didn't find a span with name %s", tt.matcher.Name)
+			require.True(t, tt.match.ended, "span wasn't ended")
+			if tt.matcher.Kind != 0 {
+				require.EqualValues(t, tt.matcher.Kind, tt.match.kind, "span kind values don't match")
+			}
+			require.EqualValues(t, matcher.Status, tt.match.status, "span status values don't match")
+			require.ElementsMatch(t, matcher.Attributes, tt.match.attrs, "span attributes don't match")
+			require.ElementsMatch(t, matcher.Links, tt.match.links, "span links don't match")
```

Comment:
@jhendrixMSFT would you prefer this to be checked only if we specified it in the matcher (similar to the span kind above) or for this to be checked in all cases?

Reply:
I think this is fine. IIRC there are legit cases where a span's `Kind` is the zero-value, so we only check it when it has an explicit value.